src/Razor/src/Microsoft.CodeAnalysis.Remote.Razor/ProjectSystem/ProjectSnapshotFactory.cs

```diff
@@ -11,7 +11,7 @@ namespace Microsoft.CodeAnalysis.Remote.Razor.ProjectSystem;
 [method: ImportingConstructor]
 internal class ProjectSnapshotFactory(DocumentSnapshotFactory documentSnapshotFactory, ITelemetryReporter telemetryReporter)
 {
-    private static readonly ConditionalWeakTable<Project, RemoteProjectSnapshot> _projectSnapshots = new();
+    private readonly ConditionalWeakTable<Project, RemoteProjectSnapshot> _projectSnapshots = new();
```

Comment:
Snuck in this change, and the above, because I noticed it was wrong. The services themselves are `Shared`, so there is already only one instance of this CWT in real life. In tests though, this just filled up with a bunch of things for no good reason.

Reply:
See my comment above. I'm uncomfortable with this change without a comment indicating that you know what you're doing and why this is OK. :smile:

If this ultimately has no impact on tests other than a larger table due to the GC not cleaning up yet, I don't believe that's a good reason to put a CWT in an instance field, IMHO.

Reply:
>  without a comment indicating that you know what you're doing

I absolutely do not. Will revert :)

This was only for tests, so definitely not justified given your (excellent) explanations